src/cpp.cc

```diff
@@ -2122,16 +2167,49 @@ void Printer::Cpp::multi_print_cyk(
   stream << indent() << *t << " t_" << track << "_n = t_" << real_track
     << "_seq.size();" << endl << endl;
 
+  bool checkpoint = ast->checkpoint && ast->checkpoint->cyk;
+
   // SMJ 2023-02-18: I've added the x: xxx loops comments, but
   // am not 100% sure if they are correct in multitrack contexts
   if (!inner.empty()) {
-    stream << indent() << "for (" << *t << " " << js << " = 0; " << js << " < "
+    if (checkpoint) {
+      /*
+         in checkpointing mode, the loop indices (e.g. t_0_i and t_0_j
+         in single-track mode; names can differ in multi-track mode)
+         are members of the out class instead of local loop variables
+         so they can be archived/loaded;
+         if they are loaded, the outer loop variable (e.g. t_0_j) starts
+         from whatever value was loaded from the checkpoint
+      */
+      stream << indent() << "for (; " << js << " < "
```

Comment:
when running through by debugger it looks like the loop variables are NOT properly initialized?!

`testdata/grammar/adpf.gap -p mfe --cyk --checkpoint`

Reply:
Thats weird. I'm initializing the loop variables in the `out::init` function, so they should be initialized before the `cyk` function is called.